src/services/fungibleTokens.js

```diff
@@ -0,0 +1,90 @@
+import * as Sentry from '@sentry/browser';
+import BN from 'bn.js';
+import * as nearApiJs from 'near-api-js';
+
+import sendJson from '../tmp_fetch_send_json';
+import { wallet } from '../utils/wallet';
+import { ACCOUNT_HELPER_URL } from '../utils/wallet';
+
+const FT_MINIMUM_STORAGE_BALANCE = '1250000000000000000000';
+const FT_STORAGE_DEPOSIT_GAS = '30000000000000';
+const FT_TRANSFER_GAS = '30000000000000';
+const FT_TRANSFER_DEPOSIT = '1';
+
+const {
+    transactions: {
+        functionCall
+    }
+} = nearApiJs;
+
+class FungibleTokens {
+    constructor() {
+        this.account = wallet.getAccountBasic();
+    }
+
+    async isStorageBalanceAvailable(contractName, accountId) {
+        return new BN((await this.checkStorageBalance(contractName, accountId)).total).gte(new BN(FT_MINIMUM_STORAGE_BALANCE));
+    }
+
+    async checkStorageBalance(contractName, accountId) {
+        return await this.account.viewFunction(contractName, 'storage_balance_of', { account_id: accountId }).catch(logError);
+    }
+
+    async transfer(contractName, amount, memo, receiver) {
+        await this.signAndSendTransaction(contractName, [
+            functionCall('ft_transfer', {
+                receiver_id: receiver,
+                amount: await this.calculateAmount(contractName, amount),
+                memo: memo
+            }, FT_TRANSFER_GAS, FT_TRANSFER_DEPOSIT)
+        ]);
+    }
+
+    async calculateAmount(contractName, amount) {
+        const { metadata } = await this.getMetadata(contractName);
```

Comment:
I think passing metadata is a better idea, I added a condition that if `contractMetadata` is no being pass, we are checking it inside the method, in case somebody would like to use it this way.

Reply:
Unless we have code that actively uses the fetching of metadata here, I think we should just require the metadata for now.  If we change the function signature to be non-positional, adding a new `contractName` argument later that can be provided _instead_ of `contractMetadata` will not be a breaking change -- but until we have a use case where we need this code to be fetching metadata, let's leave that functionality out entirely.  When/if we get to the point where we want to enable this object to do the fetching as part of this function call, we will want to drive how we implement that based on the details of why we need it to be doing the fetch -- most likely we'll want to use a caching provider of some kind at that point rather than directly call `ft_metadata` _every time someone calls transfer_, since we know the metadata of the contract can/should be cached.  Let's just cross that bridge when we have to, and rely on passing the contract metadata into this function call for now :). If someone needs to get the metadata, it's a trivial thing for them to call getMetadata _themselves_, e.g.:
```javascript
const FTMetadata = await fungibleTokenManager.getMetadata(contractName);

transfer({contractMetaData: { contractName, ...FTMetadata }, amount, receiver, memo });
```

Or if they want to, they can do what we've done with redux, and populate a cache with the result of `getMetadata()`, then pass the cached result into `transfer()`.  The key thing is that our really low-level method isn't responsible for fetching the data just to perform a transfer; the things it requires are passed to it rather than looked up dynamically.  This also makes testing the function a lot easier, since there is no dependency on the result of a `near-api-js` `functionCall` nested inside `transfer()`.

Reply:
Got it, done.